lib/posix/pthread.c

```diff
@@ -143,6 +142,31 @@ static void zephyr_thread_wrapper(void *arg1, void *arg2, void *arg3)
 	pthread_exit(NULL);
 }
 
+static bool pthread_attr_is_valid(const struct pthread_attr *attr)
+{
+	/*
+	 * FIXME: Pthread attribute must be non-null and it provides stack
+	 * pointer and stack size. So even though POSIX 1003.1 spec accepts
+	 * attrib as NULL but zephyr needs it initialized with valid stack.
+	 */
+	if (attr == NULL || attr->initialized == 0U || attr->stack == NULL ||
+	    attr->stacksize == 0) {
+		return false;
+	}
+
+	/* require a valid scheduler policy */
+	if (!valid_posix_policy(attr->schedpolicy)) {
+		return false;
+	}
+
+	/* require a valid detachstate */
+	if (!(attr->detachstate == PTHREAD_JOINABLE || attr->detachstate == PTHREAD_DETACHED)) {
```

Comment:
Do we need this here? Or is it sufficient to validate the parameter to `pthread_attr_setdetachstate`?

Reply:
I could probably change it to `__ASSERT_NO_MSG()`